buildSrc/subprojects/binary-compatibility/src/main/groovy/gradlebuild.binary-compatibility.gradle

```diff
@@ -85,54 +82,56 @@ dependencies {
         to.attribute(ARTIFACT_TYPE, 'gradle-classpath')
     }
 
-    // TODO update this to use a dependency on :distributionFull instead of 'rootProject.subprojects { }'
     rootProject.subprojects { subproject ->
-        if (subproject.name == 'distributionsDependencies') {
-            currentClasspath platform(subproject)
-        }
         plugins.withId('gradlebuild.distribution.api') {
             apiSourceFolders.from(subproject.sourceSets.main.allJava.sourceDirectories)
             if (subproject.plugins.hasPlugin("kotlin")) {
                 apiSourceFolders.from(subproject.sourceSets.main.kotlin.sourceDirectories)
             }
```

Comment:
This can probably be "fixed" in a similar way - I need to add a configuration that would select a variant that includes source directories + `GradleModuleApiAttribute.API`. Am I thinking this right?

Reply:
Yes. The variant with the sources exists because we need that in other places as well:

```
 attributes.attribute(GradleModuleApiAttribute.attribute as Attribute<GradleModuleApiAttribute>, GradleModuleApiAttribute.API)
 attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.JAVA_RUNTIME))
 attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.DOCUMENTATION))
 attribute(DocsType.DOCS_TYPE_ATTRIBUTE, objects.named("gradle-source-folders"))
```

See:
https://github.com/gradle/gradle/blob/d65bb1d033ae3066074f2982b8c7c55292793d0f/buildSrc/subprojects/jvm/src/main/kotlin/gradlebuild/unittest-and-compile.gradle.kts#L71-L87

Reply:
Looks like it is missing Kotlin sources. You can try to add them in 'unittest-and-compile.gradle.kts'. It might just work. The Javadoc and API metadata tasks will then see them, but they might just ignore the files which is fine.